### ISSUE TITLE
E0107 ( incorrect number of generic arguments) false negative 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#10351]
## Environment

* **IntelliJ Rust plugin version:** 0.4.193.5328-231-nightly
* **Rust toolchain version:** 1.70.0-nightly (88fb1b922 2023-04-10) x86_64-apple-darwin
* **IDE name and version:** CLion 2023.1.1 (CL-231.8109.222)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.ide.intentions.macros.function-like, org.rust.macros.proc

## Problem description
<img width="744" alt="image" src="https://user-images.githubusercontent.com/49211026/232016158-4742dec8-7934-42fe-a07e-5887b353b05c.png">

## Steps to reproduce
```rust
trait Tr<'a, 'b> {}

fn foo<T>(t: T)
    where
            for<'a, 'b> &'a T: Tr<'a>, // Tr<'a> <- error
{}

```

**Comment by sionzee** [comment]
I encountered a similar issue while working with the code from the windows-rs sample at [https://github.com/microsoft/windows-rs/blob/0.48.0/crates/samples/windows/create_window/src/main.rs](https://github.com/microsoft/windows-rs/blob/0.48.0/crates/samples/windows/create_window/src/main.rs). Although the code compiles and runs without any problems, my IDE (IntelliJ Rust plugin) shows an error message:

```rust
fn main() -> Result<()> {
    // ...
}
```

The error message I'm receiving is as follows:

```rust
error[E0107]: this type alias takes 1 generic argument but 2 generic arguments were supplied
  --> src\main.rs:6:14
   |
6  | fn main() -> Result<(), ()> {
   |              ^^^^^^     -- help: remove this generic argument
   |              |
   |              expected 1 generic argument
   |
note: type alias defined here, with 1 generic parameter: `T`
  --> C:\Users\jan\.cargo\registry\src\github.com-1ecc6299db9ec823\windows-0.48.0\src\core\mod.rs:41:10
   |
41 | pub type Result<T> = std::result::Result<T, Error>;
   |          ^^^^^^ -
```

Despite the error message displayed in the IDE, the code works as expected when compiled and executed. I'm sharing this issue to raise awareness of this IDE-related problem and to help others who might be facing a similar situation.
